adoc/admin-dr-backup-etcd.adoc

```diff
@@ -0,0 +1,230 @@
+
+You should also have a rotation method to clean up the unneeded snapshots over time.
+
+When there is insufficient space available during backup, pods will fail to be in `Running` state and `no space left on device` errors will show in pod logs.
+
+The below example manifest shows a binding to a local `hostPath`.
+We strongly recommend using other storage methods instead.
+====
+.. Modify the script example
++
+Replace `<STORAGE_MOUNT_POINT>` with the directory in which to store the backup.
+The directory must exist on every node in cluster.
++
+Replace `<IN_CLUSTER_ETCD_IMAGE>` with the `etcd` image used in the cluster.
+This can be retrieved by accessing any one of the nodes in the cluster and running:
++
+----
+grep image: /etc/kubernetes/manifests/etcd.yaml | awk '{print $2}'
+----
+.. Create a backup deployment
++
+Run the following script:
++
+[source,bash]
+----
+ETCD_SNAPSHOT="<STORAGE_MOUNT_POINT>/etcd_snapshot"
+ETCD_IMAGE="<IN_CLUSTER_ETCD_IMAGE>"
+MANIFEST="etcd-backup.yaml"
+
+cat << *EOF* > ${MANIFEST}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: etcd-backup
+  namespace: kube-system
+  labels:
+    jobgroup: backup
+spec:
+  template:
+    metadata:
+      name: etcd-backup
+      labels:
+        jobgroup: backup
+    spec:
+      containers:
+      - name: etcd-backup
+        image: ${ETCD_IMAGE}
+        env:
+        - name: ETCDCTL_API
+          value: "3"
+        command: ["/bin/sh"]
+        args: ["-c", "etcdctl --endpoints=https://127.0.0.1:2379 --cacert=/etc/kubernetes/pki/etcd/ca.crt --cert=/etc/kubernetes/pki/etcd/healthcheck-client.crt --key=/etc/kubernetes/pki/etcd/healthcheck-client.key snapshot save /backup/etcd-snapshot-\$(date +%Y-%m-%d_%H:%M:%S_%Z).db"]
+        volumeMounts:
+        - mountPath: /etc/kubernetes/pki/etcd
+          name: etcd-certs
+          readOnly: true
+        - mountPath: /backup
+          name: etcd-backup
+      restartPolicy: OnFailure
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      tolerations:
+      - effect: NoSchedule
+        operator: Exists
+      hostNetwork: true
+      volumes:
+      - name: etcd-certs
+        hostPath:
+          path: /etc/kubernetes/pki/etcd
+          type: DirectoryOrCreate
+      - name: etcd-backup
+        hostPath:
+          path: ${ETCD_SNAPSHOT}
+          type: Directory
+*EOF*
+
+kubectl create -f ${MANIFEST}
+----
++
+If you are using local `hostPath` and not using a shared storage device, the `etcd` backup will be created to any one of the master nodes.
+To find the node associated with each `etcd` backup run:
++
+[source,bash]
+----
+kubectl get pods --namespace kube-system --selector=job-name=etcd-backup -o wide
+----
+
+== Schedule ETCD Cluster Backup
+. Mount external storage device to all master nodes.
+This is only required if the following step is using local `hostPath` as volume storage.
+. Create Cronjob.
+.. Find the size of the database to be backed up
++
+[IMPORTANT]
```

Comment:
You had some "TODO"s in here about how to better write "Hints". The best way is to use the Admonition macro: https://asciidoctor.org/docs/user-manual/#admonition

Severity increases from TIP to WARNING. 
CAUTION and WARNING should only be used if the information presented will lead to data loss or damage to the data if ignored.

TIP is something that makes life easier
NOTE is something you should be aware of but not necesserily need to do
IMPORTANT is something you must do or consider
CAUTION might lose data
WARNING might lose machine and data